File: api/Lib_Primavera/Model/SalesOrder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace SFA_REST.Lib_Primavera.Model
{
    public class SalesOrder
    {

        public string id
        {
            get;
            set;
        }

        public string entity
        {
            get;
            set;
        }

        public int numDoc
        {
            get;
            set;
        }

        public DateTime date
        {
            get;
            set;
        }

        public double totalMerc
        {
            get;
            set;
        }

        public string serie
        {
            get;
            set;
        }

        public string salesRep
        {
            get;
            set;
        }

        public List<Model.LinhaDocVenda> LinhasDoc

        {
            get;
            set;
        }
 

    }
}

File: web/app/class/salesorder.ts
import { SalesOrderLine } from './salesorderline';

export class SalesOrder {
  id: string;
  date: Date;
  entity: string;
  numDoc: string;
  salesRep: string;
  serie: string;
  total: string;
  address: string;
  lines: SalesOrderLine[] = [];

  constructor(data: JSON) {
    this.id = data['id'];
    this.date = new Date(data['date']);
    this.entity = data['entity'];
    this.numDoc = data['numDoc'];
    this.salesRep = data['salesRep'];
    this.serie = data['serie'];
    this.total = data['totalMerc'];
    this.address = data['address'];
    if (data['LinhasDoc'] != null)
      for (let line of data['LinhasDoc'])
        this.lines.push(new SalesOrderLine(line));
  }
}